drgn_tools/dentry.py

```diff
@@ -381,24 +381,46 @@ def __file_type(mode: Object) -> str:
     return "UNKN"
 
 
-def ls(prog: Program, directory: str, count: bool = False) -> None:
+def ls(
+    prog: Program,
+    directory: str,
+    parent_dentry: Optional[Object] = None,
+    depth: int = 0,
+    count: bool = False,
+) -> None:
     """
     Print dentry children, like the ls command
     :param directory: directory to print children of
+    :param parent_dentry: dentry object of the directory to print children of
+    :param depth: If provided, it recursively prints the children of subdirectories up to the specified depth.
     :param count: when true, only print counts (not the full contents)
     """
-    dentries = dentry_for_each_child(path_lookup(prog, directory).dentry)
 
+    subdirs = []
+    dentries = dentry_for_each_child(
+        parent_dentry if parent_dentry else path_lookup(prog, directory).dentry
+    )
+
+    print(f"{directory}:")
     pos = neg = 0
     for i, dentry in enumerate(dentries):
-        path = dentry_path_any_mount(dentry).decode()
+        name = dentry.d_name.name.string_().decode()
         if dentry_is_negative(dentry):
             neg += 1
+            dentry_type = "negative"
         else:
+            dentry_type = __file_type(int(dentry.d_inode.i_mode))
+            if dentry_type == "DIR":
+                subdirs.append((f"{directory}/{name}", dentry))
             pos += 1
         if not count:
-            print(f"{i:05d} {path}")
-    print(f"{pos} positive, {neg} negative dentries")
+            print(f"\t{i:05d} {name} ({dentry_type})")
```

Comment:
I'm so sorry for only catching this now, since the PR looks great otherwise. I think the output would be much easier to scan and read if we put it in columns like below:
```suggestion
            print(f"\t{i:05d} {dentry_type:4s} {name}")
```
And then please make the `negative` text be `NEG` so that it is similar to the other dentry types.